plugins/osdn/common.go

```diff
@@ -50,15 +44,29 @@ type OsdnController struct {
 	adminNamespaces []string
 }
 
-// Called by plug factory functions to initialize the generic plugin instance
-func (oc *OsdnController) BaseInit(registry *Registry, pluginHooks PluginHooks, multitenant bool, hostname string, selfIP string) error {
+// Called by higher layers to create the plugin SDN master instance
+func NewMasterPlugin(pluginType string, osClient *osclient.Client, kClient *kclient.Client) (api.OsdnPlugin, error) {
+	if !isOsdnPlugin(pluginType) {
+		return nil, nil
+	}
+	return createPlugin(osClient, kClient, isMultitenantPlugin(pluginType), "", "")
+}
+
+// Called by higher layers to create the plugin SDN node instance
+func NewNodePlugin(pluginType string, osClient *osclient.Client, kClient *kclient.Client, hostname string, selfIP string) (api.OsdnPlugin, error) {
+	if !isOsdnPlugin(pluginType) {
+		return nil, nil
+	}
+	return createPlugin(osClient, kClient, isMultitenantPlugin(pluginType), hostname, selfIP)
```

Comment:
We could pass pluginType to createPlugin() [L52, L60] and in L93, we can evaluate if the plugin is multitenant or not in one place:
```
plugin := &OsdnController{
 	multitenant:  IsMultitenantPlugin(pluginType) ,
...
```